ament_tools/build_types/cmake.py

```diff
@@ -167,12 +177,15 @@ def _common_cmake_on_test(self, context, build_type):
         prefix = self._get_command_prefix('test', context)
         if not IS_WINDOWS:
             if has_make_target(context.build_space, 'test') or context.dry_run:
-                if MAKE_EXECUTABLE is None:
-                    raise VerbExecutionError("Could not find 'make' executable")
-                cmd = prefix + [MAKE_EXECUTABLE, 'test']
-                if 'ARGS' not in os.environ:
-                    cmd.append('ARGS="-V"')
-                yield BuildAction(cmd)
+                if CTEST_EXECUTABLE is None:
+                    raise VerbExecutionError("Could not find 'ctest' executable")
+                # invoke CTest directly in order to pass arguments
+                yield BuildAction(prefix + [
+                    CTEST_EXECUTABLE,
+                    # show all test output
+                    '-V',
+                    '--force-new-ctest-process'] +
+                    context.ctest_args)
```

Comment:
> Yes, but using the test target of make relies on the fact that the generated code internally uses the ARGS environment variable.

I don't understand what you mean by that. `ctest` uses `ARGS` directly (it's a feature of `ctest`), and I don't think any of our generated code needs to do anything with it.

Either way I don't care as long as you haven't overlooked a difference between the `make test` target and invoking `ctest` directly.